src/client/components/Inputs/Select/Indicators/Indicators.tsx

```diff
@@ -41,3 +50,31 @@ export const ClearIndicator: React.FC<ClearIndicatorProps> = (props) => {
     </components.ClearIndicator>
   )
 }
+
+export const MultiSelectOption: React.FC<OptionProps<Option>> = (props) => {
+  const { data, label } = props
+
+  const { checked, isInputIndeterminate } = useMultiSelectOptionConfig(props)
+
+  const inputRef = useRef<HTMLInputElement>(null)
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.indeterminate = isInputIndeterminate
+    }
+  }, [isInputIndeterminate])
+
+  return (
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    <components.Option {...props}>
+      <input
+        key={`${data.value}-${isInputIndeterminate}`}
```

Comment:
Yes, we need it here because the `indeterminate` attribute can't be passed directly to the input. So, if that attribute changes, the input won't be re-rendered. The `key` takes care of that by adding `isInputIndeterminate` to it.